examples/src/main/java/io/numaproj/numaflow/examples/sideinput/README.md

```diff
@@ -0,0 +1,85 @@
+# SideInput Example
+
+An example that demonstrates how to write a `sideinput` function along with a sample `User Defined function`
+which watches and uses the corresponding side input.
+
+### SideInput
+```java
+public class SimpleSideInput extends SideInputRetriever {
+    private final Config config;
```

Comment:
`Config` is a class defined under `examples/sideinput/simple` instead of `examples/sideinput/`. Consider moving the example code to `examples/sideinput/simple/README.md`. For this top-level README.md, we can just put a link to the simple side input folder as example.